simpleviewinc/keg-cli
# feat: GitHub tests workflow

### Context
* We should leverage the Keg-CLI to run commands needed in our CI environment
* This would allow us to build docker images, push updates to npm and more

### Updates
* This is the first step in setting up github workflows for the Keg-CLI
* Currently it just setups up an environment which allows the keg-cli tests to be run and pass
  * Eventually it will add the ability to run keg commands within github workflows
  * It could then be leveraged within  keg-hub, and tap repos

### Tests

**Setup**
* Fork this PR, and pull it locally

**Test 1**
* Make a NON-breaking change to one of the keg-cli files
* Run the tests locally, to ensure they work locally
* Commit and push your changes to your own fork
* Create a pull request to your own forks develop branch
  * This should kick off the github action to run the keg-cli tests
  * Ensure all tests pass within the CI environment

**Test 2**
* Do all the steps in step one, but make a breaking change to one of the tests 
* Ensure that when the tests fail, the job exits. It should not succeed if the tests fail
